lib/core/MantraData.js

```diff
@@ -0,0 +1,303 @@
+     * @returns {boolean} - True if the relationship name exists in the schema property.
+     */
+    _isId(value) {
+        return IS_ID_NUMBER(value);
+    };
+
+    /**
+     * Set the id property to the id parameter.
+     * Updates the path property and trimms the _path property.
+     * @param {string} id - The id value.
+     */
+    _setId(id) {
+        this.id = id;
+        this._trimPath(id);
+    };
+
+    /**
+     * Sets the id property to it's initial value.
+     */
+    _resetId() {
+        this.id = '';
+    };
+
+    /**
+     * Append the previous entity object to the parents property.
+     * @param {string} entity - The previous entity object.
+     */
+    _addParent(entity) {
+        this.parents.push(entity);
+    };
+
+    /**
+     * Sets the action property to the returned action based on the _path property.
+     */
+    setAction() {
+        this.action = this._getAction();
+    };
+
+    /**
+     * Gets a custom action or a default action object based on the _path property.
+     * @returns {object} - A default or custom action object.
+     */
+    _getAction() {
+        if (this._isCustomAction()) 
+            return this._getCustomAction();
+        else 
+            return this._getDefaultAction();
+    };
+
+    /**
+     * Verify if the first string in the _path parameter is a custom action.
+     * @returns {boolean} - True if a custom action matches the action name.
+     */
+    _isCustomAction() {
+        const [action = ''] = this._path.split('.');
+        const HAS_ACTIONS = this.schema.actions;
+        return HAS_ACTIONS && this.schema.actions.hasOwnProperty(action);
+    };
+
+    /**
+     * Gets the custom action object that matches the action name.
+     * @returns {object} - A custom action object.
+     */
+    _getCustomAction() {
+        const [action = ''] = this._path.split('.');
+        return this.schema.actions[action];
+    };
+
+    /**
+     * Verify if the default action should be the update action.
+     * @returns {boolean} - True if the id property is not empty.
+     */
+    _isUpdateAction() {
+        const IS_ID_EMPTY = (this.id !== '');
+        return IS_ID_EMPTY;
+    };
+
+    /**
+     * Gets a default action object based on MantraData's properties.
+     * @returns {object} - A built-in Mantra action object.
+     */
+    _getDefaultAction() {
+        if (this._isUpdateAction()) 
+            return DEFAULT_ACTIONS.update;
+        else 
+            return DEFAULT_ACTIONS.create;
+    };
+
+    /**
+     * Sets the component property to the returned component based on the _baseClass property.
+     */
+    setComponent() {
+        const component = this._baseClass.getComponent(this);
+        this.component = component;
+    };
+
+    /**
+     * Sets the endpoint property to a string joined by slash.
+     */
+    setEndpoint() {
+        const endpoint = this.getEndpoint();
+        this.endpoint = endpoint;
```

Comment:
Same suggestion as with the one at the setComponent() method

Reply:
Sure! Will do.